recipes/microprofile/all/patches/cmake.patch

```diff
@@ -0,0 +1,132 @@
++add_definitions("-DMICROPROFILE_PER_THREAD_GPU_BUFFER_SIZE=${MP_THREAD_GPU_BUFFER_SIZE}")
++add_definitions("-DMICROPROFILE_MAX_FRAME_HISTORY=${MP_MAX_FRAME_HISTORY}")
++add_definitions("-DMICROPROFILE_WEBSERVER_PORT=${MP_WEBSERVER_PORT}")
++add_definitions("-DMICROPROFILE_WEBSERVER_MAXFRAMES=${MP_WEBSERVER_MAXFRAMES}")
++add_definitions("-DMICROPROFILE_WEBSERVER_SOCKET_BUFFER_SIZE=${MP_WEBSERVER_SOCKET_BUFFER_SIZE}")
++add_definitions("-DMICROPROFILE_GPU_FRAME_DELAY=${MP_GPU_FRAME_DELAY}")
++add_definitions("-DMICROPROFILE_NAME_MAX_LEN=${MP_NAME_MAX_LENGTH}")
++add_definitions("-DMICROPROFILE_MAX_TIMERS=${MP_MAX_TIMERS}")
++add_definitions("-DMICROPROFILE_MAX_THREADS=${MP_MAX_THREADS}")
++add_definitions("-DMICROPROFILE_MAX_STRING=${MP_MAX_STRING_LENGTH}")
++add_definitions("-DMICROPROFILE_TIMELINE_MAX_TOKENS=${MP_TIMELINE_MAX_TOKENS}")
++add_definitions("-DMICROPROFILE_THREAD_LOG_FRAMES_REUSE=${MP_THREAD_LOG_FRAMES_REUSE}")
++add_definitions("-DMICROPROFILE_MAX_GROUPS=${MP_MAX_GROUPS}")
++
++if (MP_ENABLED)
++    add_definitions("-DMICROPROFILE_ENABLED=1")
++else()
++    add_definitions("-DMICROPROFILE_ENABLED=0")
++endif()
++
++if (MP_DEBUG)
++    add_definitions("-DMICROPROFILE_DEBUG=1")
++else()
++    add_definitions("-DMICROPROFILE_DEBUG=0")
++endif()
++
++if (MP_MINIZ)
++    add_definitions("-DMICROPROFILE_MINIZ=1")
++else()
++    add_definitions("-DMICROPROFILE_MINIZ=0")
++endif()
++
++if (MP_BIG_ENDIAN)
++    add_definitions("-DMICROPROFILE_BIG_ENDIAN=1")
++else()
++    add_definitions("-DMICROPROFILE_BIG_ENDIAN=0")
++endif()
++
++if (MP_GPU_TIMERS_GL OR MP_GPU_TIMERS_D3D11 OR MP_GPU_TIMERS_D3D12 OR MP_GPU_TIMERS_VULKAN)
++    add_definitions("-DMICROPROFILE_GPU_TIMERS=1")
++else()
++    add_definitions("-DMICROPROFILE_GPU_TIMERS=0")
++endif()
++
++if (MP_GPU_TIMER_CALLBACKS)
++    add_definitions("-DMICROPROFILE_GPU_TIMER_CALLBACKS=1")
++else()
++    add_definitions("-DMICROPROFILE_GPU_TIMER_CALLBACKS=0")
++endif()
++
++if (MP_GPU_TIMERS_GL)
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_GL=1")
++else()
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_GL=0")
++endif()
++
++if (MP_GPU_TIMERS_D3D11)
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_D3D11=1")
++else()
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_D3D11=0")
++endif()
++
++if (MP_GPU_TIMERS_D3D12)
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_D3D12=1")
++else()
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_D3D12=0")
++endif()
++
++if (MP_GPU_TIMERS_VULKAN)
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_VULKAN=1")
++else()
++    add_definitions("-DMICROPROFILE_GPU_TIMERS_VULKAN=0")
++endif()
++
++if (${BUILD_SHARED_LIBS})
++    add_definitions("-D_MICROPROFILE_SHARED")
++else()
++    add_definitions("-D_MICROPROFILE_STATIC")
++endif()
++
++set(SOURCES
++    microprofile.h
++    microprofile.cpp
++)
++
++add_library(${PROJECT_NAME} ${SOURCES})
++
++if (WIN32)
++    list(APPEND LIBRARIES ws2_32)
```

Comment:
FYI, you can call `target_link_libraries` as much as you'd like. CMake will automatically append the libraries.
```suggestion
+    target_link_libraries(${PROJECT_NAME} PRIVATE ws2_32)
```